test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1

```diff
@@ -687,6 +687,42 @@ Describe "TabCompletion" -Tags CI {
             $res.CompletionMatches[1].CompletionText | Should -BeExactly 'dog'
         }
 
+        It "Tab completion for ArgumentCompleter when AST is passed to CompleteInput" {
+            $script = @'
+function Test-Completion {
+    param (
+        [String]$TestVal
+    )
+}
+
+Register-ArgumentCompleter -CommandName Test-Completion -ParameterName TestVal -ScriptBlock $completer
+
+Test-Completion -TestVal
+'@
+            # Workaround to prevent editors from trimming required trailing space.
+            $script += " "
+            $scriptBl = {
+                function Test-Completion {
+                    param (
+                        [String]$TestVal
+                    )
+                }
+                [scriptblock]$completer = {
+                    param($commandName, $parameterName, $wordToComplete, $commandAst, $fakeBoundParameters)
+
+                    @('Val1', 'Val2')
+                }
+                Register-ArgumentCompleter -CommandName Test-Completion -ParameterName TestVal -ScriptBlock $completer
+            }
+            $pwsh = [PowerShell]::Create()
+            $pwsh.AddScript($scriptBl)
+            $pwsh.Invoke()
```

Comment:
This doesn't seem right. After invoking, `Test-Completion` would already been defined, and it's not the scenario you are aiming at -- `AST or Script has matching function definition`.

Reply:
My understanding of this is as follows:

The script block is executed to define the Test-Completion function and argument completer in the runspace.(As the runspace should provide the registered argument completers when in an editor).

From there, we are using the $script variable for completion input(Similar to how VSCode utilizes CompleteInput) as the issue arises when the function is defined in the $script block being passed to CompleteInput